components/event-receiver/event-input-adapters/org.wso2.carbon.event.input.adapter.sqs/src/main/java/org/wso2/carbon/event/input/adapter/sqs/internal/SQSTask.java

```diff
@@ -0,0 +1,123 @@
+        this.tenantId = tenantId;
+        this.sqs = sqs;
+        this.eventAdapterListener = eventAdapterListener;
+        this.configs = configs;
+        this.receiveMessageRequest = new ReceiveMessageRequest()
+                .withQueueUrl(configs.getQueueURL())
+                .withMaxNumberOfMessages(configs.getMaxNumberOfMessages());
+
+        if (configs.getWaitTime() != null) {
+            receiveMessageRequest = receiveMessageRequest.withWaitTimeSeconds(configs.getWaitTime());
+        }
+
+        if (configs.getVisibilityTimeout() != null) {
+            receiveMessageRequest = receiveMessageRequest.withVisibilityTimeout(configs.getVisibilityTimeout());
+        }
+    }
+
+    @Override
+    public void run() {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+            List<Message> messages = sqs.receiveMessage(receiveMessageRequest).getMessages();
+            if (messages != null) {
+                for (Message message : messages) {
+                    String eventMsg = message.getBody();
+                    sendEventToListener(eventMsg);
+                    if (configs.shouldDeleteAfterConsuming()) {
+                        delete(message);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Error occured while trying to receive messages from the queue. Hence waiting for the next " +
+                    "polling cycle.", e);
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+    }
+
+    private void sendEventToListener(String eventMsg) {
+        try {
+            eventAdapterListener.onEvent(eventMsg);
+        } catch (Exception e) {
+            /*
+                Catching unexpected exceptions occurring after onEvent(), in order to avoid
+                thread getting killed.
+            */
+            LOG.error(String.format("Error while transforming the event : %s", eventMsg), e);
+        }
+    }
+
+    private void delete(Message message) {
+        int deleteRetryCount = 0;
+        while (!deleteMessageFromQueue(message) && (deleteRetryCount < configs.getRetryCountLimit())) {
+            deleteRetryCount++;
+            try {
+                Thread.sleep(configs.getRetryInterval());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    private boolean deleteMessageFromQueue(Message message) {
+        String messageReceiptHandle = message.getReceiptHandle();
+        try {
+            DeleteMessageResult deleteMessageResult =
+                    sqs.deleteMessage(new DeleteMessageRequest(configs.getQueueURL(), messageReceiptHandle));
+            return deleteMessageResult.getSdkHttpMetadata().getHttpStatusCode() == 200;
+        } catch (Exception e) {
```

Comment:
Can't we catch specific exceptions here rather than catching generic exception

Reply:
Fixed with PR #531